### ISSUE TITLE
minpoly based simplify function

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#18907]
We should have a simplification function that simplifies algebraic numbers and algebraic functions by using minpoly. It will work well in any case where the expression reduces down to a very simple expression. This has been mentioned in some other issues (search the tracker for "minpoly"), but I don't think we have a real issue for it.

In particular, if the minimal polynomial is linear or constant, then the simplification is obvious. The resulting expression should generally be simpler. For example

```py
>>> minpoly((sqrt(x)+1)/(sqrt(x)*(x+sqrt(x))), y)
x*y - 1
```

This proves that the expression `(sqrt(x)+1)/(sqrt(x)*(x+sqrt(x)))` is equal to `1/x` (the root of the polynomial `x*y - 1` in `y`).

If the polynomial has a higher degree it is harder, because you have to make sure you keep the same root. Also the root finding algorithms might not necessarily produce a simpler expression (or even find an expression at all). 

It should be easy to start with something that works for the degree <= 1 case. Then we can expand it to work in the higher degree cases.

**Comment by asmeurer** [comment]
A concern with this for algebraic functions is that produces functions that are algebraically equivalent, but may not be functionally equivalent due to different branch cuts. For example

```py
>>> minpoly(sqrt(x**2), y)
-x + y
```

I don't know if there is a way around that, or we should just have it as a caveat.

As far as getting the right root for higher degrees, we can use something like this

```py
>>> expr = 1/(sqrt(2) + 1)
>>> p = minpoly(expr, x)
>>> p
x**2 + 2*x - 1
>>> s = solve(p)
>>> s
[-1 + sqrt(2), -sqrt(2) - 1]
>>> minpoly(expr - s[0], x)
x
>>> minpoly(expr - s[1], x)
x**2 - 8
```

This shows that `s[0]` is the root that is the same as the original expression.

**Comment by oscarbenjamin** [comment]
Is it necessary to find the roots of the monpoly or can something like `ratsimpmodprime` be used with the minpoly like in https://github.com/sympy/sympy/issues/15753#issuecomment-595999499

**Comment by oscarbenjamin** [comment]
Maybe this could also work for `RootOf` as well

**Comment by asmeurer** [comment]
Indeed, perhaps that is a better approach. We should definitely use it because it will work even for the cases where root finding does not.

I was hoping that this could also work to do something akin to sqrtdenest, but it seems our root finding algorithms don't do a great job there. 

```py
>>> sqrtdenest(sqrt(5 + 2 * sqrt(6)))
sqrt(2) + sqrt(3)
>>> minpoly(sqrt(5 + 2 * sqrt(6)), x)
x**4 - 10*x**2 + 1
>>> solve(minpoly(sqrt(5 + 2 * sqrt(6)), x))
[-sqrt(5 - 2*sqrt(6)), sqrt(5 - 2*sqrt(6)), -sqrt(2*sqrt(6) + 5), sqrt(2*sqrt(6) + 5)]
```

But I think in theory at least the roots from solve could be simpler. I don't know if ratsimpmodprime  would work in these situations.

**Comment by oscarbenjamin** [comment]
When trying out `ratsimpmodprime` I did also find that it is quite slow

**Comment by asmeurer** [comment]
I believe that's because it uses Groebner bases, which can be slow. I don't know if minpoly can be slow in more complicated cases as well, but unless I am misremembering it doesn't use Groebner.
